### PR TITLE
Add Topic Operator's Cruise Control additional metrics

### LINKED ISSUE #10124: Add Cruise Control timer metrics to the Topic Operator
### Related problem

In #9216, we created additional metrics for requests to external services (Kubernetes and Kafka). These additional metrics are disabled by default, but can be enabled to track down performance issues, or while doing load tests.

In #9483, we introduced the topic replication factor change feature, which integrates the Topic Operator with Cruise Control REST API.

### Suggested solution

It would be good to also have timer metrics for the `topic_configuration` and `user_tasks` requests. These should be present only when the additional metrics and Cruise Control integration are enabled.

### PR DESCRIPTION
**Pull request by fvaleri**
This change adds missing timers for Cruise Control's requests to the Topic Operator.

Should close #10124.

### CODE REVIEW

File: topic-operator/src/test/java/io/strimzi/operator/topic/TopicOperatorMetricsTest.java
Comment (tombentley) [inline topic-operator/src/test/java/io/strimzi/operator/topic/TopicOperatorMetricsTest.java:230]: It would be better to assert the metrics after each reconciliation.

File: topic-operator/src/main/java/io/strimzi/operator/topic/ReplicasChangeHandler.java
Comment (tombentley) [inline topic-operator/src/main/java/io/strimzi/operator/topic/ReplicasChangeHandler.java:105]: I'd guess you should also stop the time if an exception is thrown?

File: topic-operator/src/main/java/io/strimzi/operator/topic/ReplicasChangeHandler.java
Comment (tombentley) [inline topic-operator/src/main/java/io/strimzi/operator/topic/ReplicasChangeHandler.java:137]: I'd guess you should also stop the time if an exception is thrown?

**Review by tombentley — COMMENTED** [review]
Since these additional timers seem to simply be measuring the request/response duration, did you consider building metric collection into the client?

File: topic-operator/src/test/java/io/strimzi/operator/topic/TopicOperatorMetricsTest.java
Comment (fvaleri) [reply topic-operator/src/test/java/io/strimzi/operator/topic/TopicOperatorMetricsTest.java:284]: Definitely.

File: topic-operator/src/main/java/io/strimzi/operator/topic/ReplicasChangeHandler.java
Comment (fvaleri) [reply topic-operator/src/main/java/io/strimzi/operator/topic/ReplicasChangeHandler.java]: Right. Thanks for catching this.

File: topic-operator/src/main/java/io/strimzi/operator/topic/ReplicasChangeHandler.java
Comment (fvaleri) [reply topic-operator/src/main/java/io/strimzi/operator/topic/ReplicasChangeHandler.java]: Right. Thanks for catching this.

**Comment by fvaleri** [comment]
@tombentley thanks for catching the exception issue.

> did you consider building metric collection into the client

yes, and I prefer to keep it separate, as we have for the other clients

**Review by ppatierno — APPROVED** [review]
LGTM even if as not an SME on the TO and still not understanding why we decided to use `var` so heavily.

**Comment by fvaleri** [comment]
@ppatierno thanks for the review.

> still not understanding why we decided to use var so heavily

This has been raised multiple times now and from different maintainers, so I would like to address it.

The `var` Java keyword (type inference) is just syntactic sugar, that allows you to reduce the amount of boilerplate code. It can make the code more readable by eliminating redundant information, but it can also make code less readable by eliding useful information.

It's usage is restricted to local variables with initializers and for loops. Whenever you new something, the new statement usually makes the type obvious. Whenever you use certain static factories, such as `List.of("a", "b", "c")`, it can make sense as well. Whenever the type is non-obvious or the scope is large, then I agree that it shouldn't be used. See the OpenJDK style guideline: https://openjdk.org/projects/amber/guides/lvti-style-guide

I don't have a strong opinion, although I found it handy when dealing with streams, so I would say let's focus on what we think it's best for the Strimzi project, and makes the review process easier. 

I think @scholzj and @ppatierno don't like it, but I would also like to hear from @tombentley, which originally introduced it. Do you think we should raise a separate PR to get rid of it and maybe set some checkstyle rule?

**Review by tombentley — APPROVED** [review]
LGTM, thanks @fvaleri

**Comment by tombentley** [comment]
@ppatierno 

> still not understanding why we decided to use `var` so heavily

I don't think anyone took a conscious decision: "Let's use `var` heavily". 

I think that the guidelines that @fvaleri linked to are pretty good, and I wouldn't object if you used them (when reviewing code) to avoid specific instances where `var` is unclear. But as those guidelines point out, there are code statements where  explicit types are unhelpful, so we (as reviewers) do need to be specific, rather than saying "too much `var`".

**Pull request merged**
